angular/src/lib/tooltip/tooltip-container.component.ts

```diff
@@ -0,0 +1,61 @@
+import { Component, Input,  TemplateRef, ElementRef} from '@angular/core';
+
+
+
+@Component({
+  selector: 'md-tooltip-container',
+  styleUrls: ['tooltip-container.scss'],
+  template: `
+  <div class="md-tooltip md-event-overlay md-event-overlay--arrow">
+    <div class="md-event-overlay__arrow" ></div>
+    <div  class="md-event-overlay__children">
+        <md-tooltip #mdtooltip id={{this.id}} text={{this.text}} [tooltipTemplate]="tooltipTemplate"></md-tooltip>
+    </div>
+  </div>
+  `
+})
+export class TooltipContainerComponent {
+
+  /** @prop Sets text in the tooltip */
+  @Input() text: string;
+
+   /** @prop Sets a component in the tooltip */
+  @Input() tooltipTemplate?: TemplateRef<any>;
+
+  /** @prop Sets direction of the tooltip | null */
+  @Input()
+  set direction(direction: string) {
+    this.elementRef.nativeElement.children[0].classList.add(`md-event-overlay--${direction}`);
+  }
+
+  /** @prop Sets a background color for the tooltip | null */
+  @Input()
+  set lightBackground(lightBackground: boolean) {
+    if ( lightBackground ) {
+      // md-popover has the light background
+      this.elementRef.nativeElement.children[0].classList.remove(`md-tooltip`);
+      this.elementRef.nativeElement.children[0].classList.add(`md-popover`);
+    }
+  }
+
+  /** @prop Sets whether or not to show the arrow on the tooltip  */
+  @Input()
+  set showArrow(showArrow: string) {
+    if ( showArrow === 'false' ) {
+      this.elementRef.nativeElement.children[0].classList.remove(
+        `md-event-overlay--arrow`
+      );
+      this.elementRef.nativeElement.children[0].children[0].classList.remove(
+        `md-event-overlay__arrow`
+      );
+    }
+  }
+
+   /** @prop Sets the id for screen reader */
+  @Input() id: string;
+
+    constructor(private elementRef: ElementRef) {
+    }
+
```

Comment:
we can remove these empty lines here and close out the constructor {} on 1 line.